incubator/hnc/internal/validators/hncconfig_test.go

```diff
@@ -176,6 +177,11 @@ func TestPropagateConflict(t *testing.T) {
 		name:         "Objects with the same name existing in namespaces that one is an ancestor of the other would have overwriting conflict",
 		inNamespaces: "ab",
 		allow:        false,
+	}, {
+		name:         "Should not cause a conflict if the object in the parent namespace has an exceptions selector that choose not to propagate to the conflicting child namespace",
```

Comment:
This is good but I'd still like to see more, I think. E.g. imagine the structure `a -> b -> c`, where `a` and `b` both have an object of the same name, but the object in `a` has the `none` selector. In that case, we should get a conflict with `b` but not with `a`. Are we able to test that?

Note that this would require changing line 195 where the test forest is always `"-aa"`. The reason `foresttest.Create` accepts such a concise string is to make it easier for each testcase to have a completely different tree structure.

Reply:
Added this testcase